src/commands/functions-config-export.ts

```diff
@@ -0,0 +1,180 @@
+        `${clc.bold(pInfo.projectId)}:\n\t${result.missing.join("\n\t")}`
+    );
+
+    const confirm = await promptOnce(
+      {
+        type: "confirm",
+        name: "skip",
+        default: true,
+        message: `Continue without importing configs from project ${pInfo.projectId}?`,
+      },
+      {}
+    );
+
+    if (!confirm) {
+      throw new FirebaseError("Command aborted!");
+    }
+  }
+}
+
+async function promptForPrefix(errMsg: string): Promise<string> {
+  logWarning("The following configs keys could not be exported as environment variables:\n");
+  logWarning(errMsg);
+  return await promptOnce(
+    {
+      type: "input",
+      name: "prefix",
+      default: "CONFIG_",
+      message: "Enter a PREFIX to rename invalid environment variable keys:",
+    },
+    {}
+  );
+}
+
+async function copyFilesToDir(srcFiles: string[], destDir: string): Promise<string[]> {
+  const destFiles = [];
+  for (const file of srcFiles) {
+    const targetFile = path.join(destDir, path.basename(file));
+    if (fs.existsSync(targetFile)) {
+      const overwrite = await promptOnce(
+        {
+          type: "confirm",
+          name: "overwrite",
+          default: false,
+          message: `${targetFile} already exists. Overwrite file?`,
+        },
+        {}
+      );
+      if (!overwrite) {
+        logBullet(`Skipping ${targetFile}`);
+        continue;
+      }
+    }
+    fs.copyFileSync(file, targetFile);
+    destFiles.push(targetFile);
+  }
+  return destFiles.filter((f) => f.length > 0);
+}
+
+export default new Command("functions:config:export")
+  .description("Export environment config as environment variables in dotenv format")
+  .before(requirePermissions, [
+    "runtimeconfig.configs.list",
+    "runtimeconfig.configs.get",
+    "runtimeconfig.variables.list",
+    "runtimeconfig.variables.get",
+  ])
+  .before(requireConfig)
+  .before((options) => {
+    if (options.nonInteractive) {
+      return Promise.reject(
+        new FirebaseError("This command cannot run in non-interactive mode", {
+          exit: 1,
+        })
+      );
+    }
+    return Promise.resolve();
+  })
+  .action(async (options: any) => {
+    let pInfos = configExport.getProjectInfos(options);
+    checkReservedAliases(pInfos);
+
+    logBullet(
+      "Importing functions configs from projects [" +
+        pInfos.map(({ projectId }) => `${clc.bold(projectId)}`).join(", ") +
+        "]"
+    );
+
+    await configExport.hydrateConfigs(pInfos);
+    await checkRequiredPermission(pInfos);
+    pInfos = pInfos.filter((pInfo) => pInfo.config);
+
+    logger.debug(`Loaded function configs: ${JSON.stringify(pInfos)}`);
+    logBullet(`Importing configs from projects: [${pInfos.map((p) => p.projectId).join(", ")}]`);
+
+    let prefix = "";
+    while (true) {
+      const errMsg = configExport.hydrateEnvs(pInfos, prefix);
+
+      if (errMsg.length == 0) {
+        break;
```

Comment:
This seems like a recipe for unhandled exceptions leading to infinite loops. Can we make this explicitly a two-chances-or-abort thing?

Reply:
Good call. Fixed.